Asset Cloning

tbrumm opened the pull request:
Hi Bradley,
hope the code is fine so far and you like it

Torsten

tbrumm committed:
Add Clone Asset to Menu

Add Clone Asset tot he Menu

tbrumm committed:
Clone an Asset

Clone an Asset fromexsisting Asset

tbrumm committed:
Create Asset_Local.pm

Subroutine to clone an asset

btb commented:
Thanks Torsten,
There is cloning code in Asset/Display.html, which works, but is incomplete. Could you either update that code or remove it? Also the Clone method should just go into Asset.pm

Brad

tbrumm commented:
Hi Brad,
is there code already? OK, will check it and sent another Pull request.

Thanks

Torsten

tbrumm committed:
Moved from local to general asset.pm

moved code from asset_local.pm to general asset.pm

tbrumm committed:
Delete Asset_Local.pm

Moved to general File

tbrumm commented:
Hi Brad,
changed the sub routine from Asset_local.pm to Asset.pm, but this part: (There is cloning code in Asset/Display.html) i cant find inside that file....can you pls point me to this? (possibly i'm not wake enough).

btb commented:
My mistake. It should be in Asset/Create.html
It's closely modeled on the cloning code in Ticket/Create.html

RT's _Overlay.pm mechanism was never meant to be used by distributed
extensions, and causes problems when used that way. Instead the code should
be loaded when the extension itself is loaded.

-Brad

On Sep 3, 2014, at 11:57 PM, Torsten Brumm notifications@github.com wrote:

Hi Brad,
changed the sub routine from Asset_local.pm to Asset.pm, but this part:
(There is cloning code in Asset/Display.html) i cant find inside that
file....can you pls point me to this? (possibly i'm not wake enough).

—
Reply to this email directly or view it on GitHub
https://github.com/AssetTracker/rt-extension-assettracker/pull/68#issuecomment-54417438
.

tbrumm committed:
Removed Asset Cloning Part from Create.html

Clone Asset Parts removed from Create HTML

Uncommented Transaction Customfields also cleaned from Code

tbrumm commented:
OK, removed from Create HTML all Clone Parts, no idea where they are from.

btb commented:
After looking again, I think I mostly like the old method better - I've been using it for years already, it's just not exposed in the menus. This callback to Elements/Tabs/Privileged works, we could just make it part of the default menu. 

```
<%perl>
my $request_path = $HTML::Mason::Commands::r->path_info;

    if ( $request_path =~ m{^/AssetTracker/Asset/} ) {
        if ( ( $DECODED_ARGS->{'id'} || '' ) =~ /^(\d+)$/ ) {
            my $id  = $1;
            my $Asset = RTx::AssetTracker::Asset->new( $session{'CurrentUser'} );
            $Asset->Load($id);

            my $actions = PageMenu()->child( actions => title => loc('Actions'), sort_order  => 95 );
            $actions->child( 'clone' =>
                title => 'Clone',
                path => "/AssetTracker/Asset/Create.html?Type=" . $Asset->Type . "&CloneAsset=" . $Asset->id,
            );
        }
    }
</%perl>
```
